sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.getCompletedApplications.e2e-spec.ts

```diff
@@ -0,0 +1,320 @@
+                .offering.studyEndDate,
+            fullName: getUserFullName(
+              applicationWithApprovedApplicationOfferingChange.student.user,
+            ),
+            status:
+              approvedApplicationOfferingChange.applicationOfferingChangeRequestStatus,
+            dateCompleted:
+              approvedApplicationOfferingChange.assessedDate.toISOString(),
+          },
+        ],
+        count: 3,
+      });
+  });
+
+  it("Should return only matching completed application offering changes for the location when searched.", async () => {
+    // Arrange
+
+    // Student 1 has an approved application offering change request for the institution.
+    const approvedApplicationOfferingChange =
+      await saveFakeApplicationOfferingRequestChange(
+        db,
+        {
+          institutionLocation: collegeFLocation,
+        },
+        {
+          initialValues: {
+            applicationOfferingChangeRequestStatus:
+              ApplicationOfferingChangeRequestStatus.Approved,
+          },
+        },
+      );
+    // Student 2 has an approved with student application offering change request with a different institution.
+    const approvedApplicationOfferingChangeForDifferentLocation =
+      await saveFakeApplicationOfferingRequestChange(db, undefined, {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.Approved,
+        },
+      });
+    // Student 3 has a declined by student application offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.DeclinedByStudent,
+        },
+      },
+    );
+    // Student 4 has a declined by SABC offering change request.
+    await saveFakeApplicationOfferingRequestChange(
+      db,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        initialValues: {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+        },
+      },
+    );
+
+    approvedApplicationOfferingChange.assessedDate = new Date();
+
+    await db.applicationOfferingChangeRequest.save(
+      approvedApplicationOfferingChange,
+    );
+
+    const applicationWithApprovedApplicationOfferingChange =
+      approvedApplicationOfferingChange.application;
+    const applicationWithApprovedApplicationOfferingChangeForDifferentLocation =
+      approvedApplicationOfferingChangeForDifferentLocation.application;
+
+    applicationWithApprovedApplicationOfferingChange.student.user.firstName =
+      "TestStudent";
+    // Student 2 belong to different institution but have same name.
+    applicationWithApprovedApplicationOfferingChangeForDifferentLocation.student.user.firstName =
+      "TestStudent";
```

Comment:
Could it have slightly different names to test partial matching?

Reply:
@andrepestana-aot Here my intention was, to two students with the same name but in different institutions. are you suggesting adding one more student for the same institution with the same name and returning both students?

Reply:
yes, or search for part of the name. Having another would be better. e.g.:
Student1: "TestStudentSuffix" 
Student2: "PrefixTestStudent" 

Both should be found if we look for " TestStudent"

Reply:
Got it. I liked the mentioned scenario. Added